src/lib/stack-factory.ts

```diff
@@ -0,0 +1,168 @@
+import { Command, ux } from "@oclif/core";
+import { input, confirm } from "@inquirer/prompts";
+import { Endpoint, ShellConfig } from "./config";
+import { searchSelect } from "./search-select";
+import FaunaClient, { QuerySuccess } from "./fauna-client";
+
+export interface AddStackParams {
+  endpoint?: string;
+  database?: string;
+  name?: string;
+  default?: boolean;
+  nonInteractive?: boolean;
+}
+
+export class StackFactory {
+  cmd: Command;
+  config: ShellConfig;
+
+  constructor(cmd: Command, config: ShellConfig) {
+    this.cmd = cmd;
+    this.config = config;
+  }
+  /**
+   *
+   * @param cmd - used to send info/errors to the user.
+   * This method will be executed from the project init as well as stack add method
+   * @param config
+   */
+  async addStack(params?: AddStackParams) {
```

Comment:
this was all pulled fromt he AddStack command so that it can be used by that command as well as project init one.